examples/shell/shell_common/cmd_ping.cpp

```diff
@@ -137,10 +144,11 @@ CHIP_ERROR SendEchoRequest(streamer_t * stream)
     CHIP_ERROR err = CHIP_NO_ERROR;
 
     Messaging::SendFlags sendFlags;
-    const char kRequestFormat[] = "Echo Message %" PRIu64 "\n";
+    uint32_t size                   = gPingArguments.GetEchoReqSize();
+    const char kRequestFormat[size] = "Echo Message %" PRIu64 "\n";
```

Comment:
This seems risky to me, as it allocates an array on the stack that may have an extremely large or small size (e.g. smaller than assigned text).

Reply:
Acked, we should allocate the buffer from heap instead of stack.